packages/zowe-explorer-api/src/profiles/ProfilesCache.ts

```diff
@@ -127,9 +128,8 @@ export class ProfilesCache {
 
     /**
      * Updates profile in allProfiles array and if default updates defaultProfileByType
-     *
+     * @deprecated Use `updateCachedProfile` instead
```

Comment:
since this is in maintenance I don't believe we should be deprecating and creating new APIs.  if the new API is the solution we should find a way for the existing API take it's passed information and handle accordingly with new method.

Reply:
I realized that the issue https://github.com/zowe/zowe-explorer-vscode/issues/3112 was specific to V3, so have reverted the addition of `updateCachedProfile` method on the V2 branch to avoid breaking changes.